macaroons: update to new macaroons pkg version

Roasbeef opened the pull request:
In this commit we update to the new version of the macaroons package
that doesn't require the caller to use the default implementation for
the root key store. Instead, the caller can pass in an implementation,
and based on that we'll change our behavior.

#### Pull Request Checklist

- [ ] PR is opened against correct version branch.
- [ ] Version compatibility matrix in the README and minimal required version
      in `lnd_services.go` are updated.
- [ ] Update `macaroon_recipes.go` if your PR adds a new method that is called
  differently than the RPC method it invokes.

Roasbeef commented on go.mod:
```diff
@@ -1,19 +1,20 @@
 
 go 1.16
+
+replace github.com/lightningnetwork/lnd => github.com/roasbeef/lnd v0.2.1-alpha.0.20220507000000-660f73377f6f
```
Will be removed once the lnd PR is merged.

guggero commented on macaroon_service.go:
```diff
@@ -170,40 +150,27 @@ func NewMacaroonService(cfg *MacaroonServiceConfig) (*MacaroonService, error) {
 func (ms *MacaroonService) Start() error {
-	// Open the database.
-	var err error
-	ms.db, err = kvdb.GetBoltBackend(&kvdb.BoltBackendConfig{
```
Can we extract this into a convenience function please? Otherwise we need to duplicate this code in all projects (loop/pool/faraday/litd).

guggero commented on macaroon_service_test.go:
```diff
@@ -24,14 +25,21 @@ func TestMacaroonServiceMigration(t *testing.T) {
+		DBTimeout:  defaultDBTimeout,
+	})
+	require.NoError(t, err)
+	rks, err := macaroons.NewRootKeyStorage(db)
```
The convenience function mentioned above would be handy here.

Roasbeef replied:
Added!

guggero reviewed (approved):
LGTM, just one minor request.

Roasbeef commented:
SGTM re extracting them, I have a new daemon using this fully working, so can confirm all the changes allow one to import this and have it handle the initial macaroon creation and rook key creation as well.

Roasbeef commented:
Updated to target the 0.15 branch since that has everything the 0.16 one did. I also updated the permissions file, to then add `ImportPublicKey` as well.
